pkg/vsphere/extraconfig/keys.go

```diff
@@ -161,7 +161,7 @@ func calculateScopeFromKey(key string) []string {
 	return scopes
 }
 
-func calculateKeyFromField(field reflect.StructField, prefix string, depth recursion) (string, recursion) {
+func calculateKeyFromField(log logrus.FieldLogger, field reflect.StructField, prefix string, depth recursion) (string, recursion) {
```

Comment:
This basically requires either a choice between encode/decode logger or collapsing those back into one. I'm inclined to leave it as is then swap it out for trace operation asap.